dev/PushNotifications/PushNotifications.idl

```diff
@@ -0,0 +1,134 @@
+Boolean Handled;
+};
+
+[flags]
+enum PushNotificationRegistrationKind
+{
+    PushTrigger = 0x1, // Registers a Push Trigger with Background Infrastructure
+    ComActivator = 0x2, // Registers the Project Reunion Background Task component as an InProc COM server 
+};
+
+// An abstraction over the activation Registration flow
+runtimeclass PushNotificationActivationInfo
+{
+    // Initialize using a RegistrationKind and optionally defined parameters like manifest defined activatorId
+    // 1) If kind = PushTrigger is specified, only the Push Trigger will be Registered with Background Infra
+    // 2) If kind = ComActivator is specified, the Project Reunion Background Task component will be Registered as an InProc COM server
+    PushNotificationActivationInfo(PushNotificationRegistrationKind kind, Guid taskClsid);
+
+// The CLSID associated with the Client COM server that Project Reunion will activate
+Guid TaskClsid{ get; };
+
+PushNotificationRegistrationKind Kind{ get; };
+
+// The conditions under which Push Triggers would execute
+Windows.ApplicationModel.Background.IBackgroundCondition[] GetConditions();
+void SetConditions(Windows.ApplicationModel.Background.IBackgroundCondition[] conditions);
+};
+
+enum PushNotificationChannelStatus
+{
+    InProgress, // The request is in progress and there is no retry operation
+    InProgressRetry, // The request is in progress and is in a backoff retry state. Check ExtendedError for HRESULT for retryable error.
+    CompletedSuccess, // The request completed successfully
+    CompletedFailure, // The request failed with some critical internal error. Check ExtendedError for HRESULT
+};
+
+// The PushNotificationChannel Progress result
+struct PushNotificationCreateChannelStatus
+{
+    // The last extended error we failed Channel requests on that caused the inprogress retry status. S_OK if this is just progress status.
+    HRESULT extendedError;
+
+    // Either InProgress or InProgressRetry status
+    PushNotificationChannelStatus status;
+
+    // Total Retries so far
+    UInt32 retryCount;
+};
+
+runtimeclass PushNotificationChannel
+{
+    PushNotificationChannel(Windows.Networking.PushNotifications.PushNotificationChannel channel);
+
+// The Channel Uri for app to Post a notification to.
+Windows.Foundation.Uri Uri { get; };
+
+// Expiration of the Channel
+Windows.Foundation.DateTime ExpirationTime { get; };
+
+// Unsubscribes the channel
+void Close();
+
+// In-memory Event handler for Push Notifications
+event Windows.Foundation.TypedEventHandler<PushNotificationChannel, PushNotificationReceivedEventArgs> PushReceived;
+}
+
+runtimeclass PushNotificationCreateChannelResult
+{
+    PushNotificationCreateChannelResult(
+        PushNotificationChannel channel,
+        HRESULT extendedError,
+        PushNotificationChannelStatus status);
+
+// The Push channel associated with the Result. Null if InProgress or completion failed
```

Comment:
Also null if InProgressRetry. Probably should just say "Valid only if CompletionSuccess."